File: iOS/Chill With Me/Models/UserProfile.swift
//
//  UserProfile.swift
//  Chill With Me
//
//  Created by Phillip Hoang on 2018-09-19.
//  Copyright © 2018 IC Heart Technologies. All rights reserved.
//

import Foundation

class UserProfile: Codable {
    
    var userName: String
    var yourCurrentStatus: YourCurrentStatus
    var completeFriendList: FriendGroup
    var listOfFriendGroups: [FriendGroup]
    var friendRequestList: [String]
    var refreshToken: String
    
    init(userName: String, refreshToken: String, friendGroupList: [FriendGroup], friendRequests: [String], status: YourCurrentStatus, friendList: FriendGroup) {
        self.userName = userName
        self.refreshToken = refreshToken
        self.listOfFriendGroups = friendGroupList
        self.friendRequestList = friendRequests
        self.yourCurrentStatus = status
        self.completeFriendList = friendList
    }
    
    init() {
        self.userName = ""
        self.refreshToken = ""
        self.listOfFriendGroups = [FriendGroup]()
        self.friendRequestList = [String]()
        self.yourCurrentStatus = YourCurrentStatus.init(statusType: "NA", statusDescription: "Not available", timeUntil: "0000000000", listOfGroupIds: [String](), listOfFriends: [String]())
        self.completeFriendList = FriendGroup.init()
    }
    
    func hasGroupsWithNotificationsOn() -> Bool {
        for group in self.listOfFriendGroups {
            if group.notificationsOn {
                return true
            }
        }
        return false
    }
    
    func getNumberOfFriendRequests() -> Int {
        return self.friendRequestList.count
    }
}

File: Android/java/r2chill/Model/UserProfile.java
package example.r2chill.Model;

import java.util.ArrayList;

public class UserProfile {
    private String userName;
    private YourCurrentStatus yourCurrentStatus;
    private FriendGroup completeFriendList;
    private ArrayList<FriendGroup> listOfFriendGroups;
    private ArrayList<String> friendRequestList;
    private String refreshToken;

    public UserProfile(String name, String newRefreshToken) {
        this.userName = name;
        this.refreshToken = newRefreshToken;
        this.listOfFriendGroups = new ArrayList<FriendGroup>();
        this.friendRequestList = new ArrayList<String>();
        this.yourCurrentStatus = new YourCurrentStatus();
        this.completeFriendList = new FriendGroup();
    }

    public UserProfile(String name, String newRefreshToken, ArrayList<FriendGroup> friendGroupList,
                       ArrayList<String> friendRequests, YourCurrentStatus status,
                       FriendGroup friendList) {
        this.userName = name;
        this.refreshToken = newRefreshToken;

        this.listOfFriendGroups = friendGroupList;
        this.friendRequestList = friendRequests;

        this.yourCurrentStatus = status;
        this.completeFriendList = friendList;
    }

    public UserProfile() {
        this.userName = "";
        this.refreshToken = "";
        this.listOfFriendGroups = new ArrayList<FriendGroup>();
        this.friendRequestList = new ArrayList<String>();
        this.yourCurrentStatus = new YourCurrentStatus();
        this.completeFriendList = new FriendGroup();
    }

    public FriendGroup getFriendListFromName(String name) {
        for (FriendGroup friendList : listOfFriendGroups) {
            if (friendList.getFriendListName().equals(name)) {
                return friendList;
            }
        }
        return null;
    }

    public boolean hasGroupsWithNotificationsOn() {
        for (FriendGroup group : this.listOfFriendGroups) {
            if (group.isNotificationsOn()) {
                return true;
            }
        }
        return false;
    }

    public int getNumberOfFriendRequests() {
        return friendRequestList.size();
    }

    public void setUserName(String username) {
        this.userName = username;
    }
    public String getUserName() {
        return this.userName;
    }

    public void setRefreshToken(String newRefreshToken) {
        this.refreshToken = newRefreshToken;
    }
    public String getRefreshToken() {
        return this.refreshToken;
    }

    public void setListOfFriendGroups(ArrayList<FriendGroup> friendGroups) {
        this.listOfFriendGroups = friendGroups;
    }
    public ArrayList<FriendGroup> getListOfFriendGroups() {
        return this.listOfFriendGroups;
    }

    public void setFriendRequestList(ArrayList<String> requestList) {
        this.friendRequestList = requestList;
    }
    public ArrayList<String> getFriendRequestList() {
        return this.friendRequestList;
    }

    public void setYourCurrentStatus(YourCurrentStatus status) {
        this.yourCurrentStatus = status;
    }
    public YourCurrentStatus getYourCurrentStatus() {
        return yourCurrentStatus;
    }

    public void setCompleteFriendList(FriendGroup friendList) {
        this.completeFriendList = friendList;
    }
    public FriendGroup getCompleteFriendList() {
        return this.completeFriendList;
    }
}
